attributes/default.rb

```diff
@@ -6,15 +6,26 @@
     "#{default['osl-docker']['package']['version']}-#{default['osl-docker']['package_release']}"
   default['osl-docker']['package']['package_name'] = 'docker-ce'
   default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+  default['osl-docker']['package']['setup_docker_repo'] = true
+  default['osl-docker']['package']['setup_docker_repo'] = false if node['platform_version'].to_i >= 8
+
 when 'debian'
-  default['osl-docker']['package']['package_name'] = 'docker-ce'
-  default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
-  default['osl-docker']['package']['version'] = '5:18.09.2'
-  default['osl-docker']['package']['package_version'] = '5:18.09.2~3-0~debian-stretch'
+  default['osl-docker']['package']['setup_docker_repo'] = true
+  case node['platform_version'].to_i
+  when 9
+    default['osl-docker']['package']['package_name'] = 'docker-ce'
+    default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+    default['osl-docker']['package']['version'] = '5:18.09.2'
+    default['osl-docker']['package']['package_version'] = '5:18.09.2~3-0~debian-stretch'
+  when 10
+    default['osl-docker']['package']['package_name'] = 'docker-ce'
+    default['osl-docker']['package_cli_name'] = 'docker-ce-cli'
+    default['osl-docker']['package']['version'] = '5:18.09.2'
+    default['osl-docker']['package']['package_version'] = '5:18.09.2~3-0~debian-buster'
+  end
+
 end
 case node['kernel']['machine']
-when 'x86_64'
-  default['osl-docker']['package']['setup_docker_repo'] = false if node['platform_version'].to_i >= 8
```

Comment:
Why was this touched?

Reply:
It set that variable to false for debian 9 and 10 then they wouldn't converge. I think it was only ment for centos 8.